buildingmotif/dataclasses/shape_collection.py

```diff
@@ -319,7 +319,10 @@ def gensym():
     tc_clauses = [
         f"?target rdf:type/rdfs:subClassOf* {tc.n3()} .\n" for tc in targetClasses  # type: ignore
     ]
-    clauses += " UNION ".join(tc_clauses)
+    if len(tc_clauses) > 1:
+        clauses += " UNION ".join(f"{{ {tc_clause} }}" for tc_clause in tc_clauses)
+    elif len(tc_clauses) == 1:
+        clauses += tc_clauses[0]
```

Comment:
Does this need error handling with a final `else` to handle `len(tc_clauses) < 1`?

Reply:
It doesn't. I've clarified this, and reorganized the code a bit to make the logic easier to follow